File: FirServer/FirSango/Data/Tables/GlobalConfigTable.cs
using System;
using System.Collections.Generic;
using UnityEngine;

namespace FirCommon.Data
{
	[Serializable]
	public class GlobalConfigTable
	{
		public string name;

		private Dictionary<string, GlobalConfigTableItem> dics = null;
		private List<GlobalConfigTableItem> items = new List<GlobalConfigTableItem>();

		public List<GlobalConfigTableItem> Items
		{
			get {
				return items;
			}
		}

		public void Initialize()
		{
			dics = new Dictionary<string, GlobalConfigTableItem>();
			foreach (GlobalConfigTableItem item in items)
			{
				dics.Add(item.id, item);
			}
		}

		public List<GlobalConfigTableItem> GetItems()
		{
			return items;
		}

		public void AddItem(GlobalConfigTableItem item)
		{
			items.Add(item);
		}

		public GlobalConfigTableItem GetItemByKey(string key)
		{
			GlobalConfigTableItem item = null;
			if (dics.ContainsKey(key))
			{
				dics.TryGetValue(key, out item);
			}
			return item;
		}
	}

	[Serializable]
	public class GlobalConfigTableItem
	{
    	public string id;
    	public string value;
	}
}

File: FirServer/FirServer/src/com/tables/Tables/GlobalConfigTable.java
package com.tables.Tables;

import java.util.Map;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class GlobalConfigTable
{
	public String name;		
	private List<GlobalConfigTableItem> items = new ArrayList<GlobalConfigTableItem>();
	
	private Map<String, GlobalConfigTableItem> dics = null;


	public void Initialize()
	{
		dics = new HashMap<String, GlobalConfigTableItem>();
		for (GlobalConfigTableItem item : items)
		{
			dics.put(item.id, item);
		}
	}

	public List<GlobalConfigTableItem> GetItems()
	{
		return items;
	}

	public void AddItem(GlobalConfigTableItem item)
	{
		items.add(item);
	}

	public GlobalConfigTableItem GetItemByKey(String key)
	{
        GlobalConfigTableItem item = null;
        if (dics.containsKey(key))
        {
            item = dics.get(key);
        }
        return item;
	}
}